### DISCUSSION TITLE
Why does updateSpecSize() start with waiting one second?

### DISCUSSION THREAD

**Discussion by guettli** [topolvm/topolvm#901] [Q&A] [answered]
I am looking at that code, and don't understand it:

```go
// updateSpecSize updates .Spec.Size of LogicalVolume.
func (s *LogicalVolumeService) updateSpecSize(ctx context.Context, volumeID string, size *resource.Quantity) error {
	for {
		select {
		case <-ctx.Done():
			return ctx.Err()
		case <-time.After(1 * time.Second):
		}
```

[source](https://github.com/topolvm/topolvm/blob/c29402b76847462d46ada71b790962dee1e405ce/internal/driver/internal/k8s/logicalvolume_service.go#L354-L355)

If a conflict-error happens, then waiting makes sense. But the code waits before the first attempt.

I guess you have your reasons to do that. Can you help me understand that better?

**Comment by jakobmoellerdev** [comment, marked as answer]
Hi @guettli thanks for taking a look. I actually just tried this and it seems like a simple oversight. I dont believe there is a reason for this.

 we have the same pattern here. https://github.com/topolvm/topolvm/blob/c29402b76847462d46ada71b790962dee1e405ce/internal/driver/internal/k8s/logicalvolume_service.go#L318-L323

**Reply by guettli** [reply]
Btw, this method is handy

https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict

Will you update the code or should I create a pr?

**Reply by jakobmoellerdev** [reply]
I have a PR up in https://github.com/topolvm/topolvm/pull/903/files but I need to recheck it because I think one of the updates now misbehaves and causes an extension failure. There is one occassion where RetryOnConflict could be used but its only saving one method is not really that helpful compared to normal retry.
